packages/cw-croncat-core/src/types.rs

```diff
@@ -111,17 +111,39 @@ impl BoundaryValidated {
     ) -> Result<Self, CoreError> {
         if let Some(boundary) = boundary {
             match (interval, boundary) {
-                (Interval::Cron(_), Boundary::Time { start, end }) => Ok(Self {
-                    start: start.map(|start| start.nanos()),
-                    end: end.map(|end| end.nanos()),
-                }),
+                (Interval::Cron(_), Boundary::Time { start, end }) => match (start, end) {
+                    (Some(s), Some(e)) => {
+                        if s.nanos() >= e.nanos() {
+                            return Err(CoreError::InvalidBoundary {});
+                        }
+                        Ok(Self {
+                            start: start.map(|start| start.nanos()),
+                            end: end.map(|end| end.nanos()),
+                        })
+                    }
+                    _ => Ok(Self {
+                        start: start.map(|start| start.nanos()),
+                        end: end.map(|end| end.nanos()),
+                    }),
+                },
                 (
                     Interval::Once | Interval::Immediate | Interval::Block(_),
                     Boundary::Height { start, end },
-                ) => Ok(Self {
-                    start: start.map(Into::into),
-                    end: end.map(Into::into),
-                }),
+                ) => match (start, end) {
+                    (Some(s), Some(e)) => {
+                        if s.u64() >= e.u64() {
+                            return Err(CoreError::InvalidBoundary {});
```

Comment:
In case of the block isn't it ok to be equal?
```suggestion
                        if s.u64() > e.u64() {
                            return Err(CoreError::InvalidBoundary {});
```